devito/tools/data_structures.py

```diff
@@ -660,6 +660,31 @@ def __hash__(self):
         return self._hash
 
 
+class MemoryEstimate(frozendict):
+    """
+    An immutable wrapper for a memory estimate, showing the
+    various values.
+
+    TODO: Finish this docstring
+    """
+
+    def __init__(self, *args, **kwargs):
+        self._name = kwargs.pop('name', 'memory_estimate')
+        super().__init__(*args, **kwargs)
+
+    @property
+    def name(self):
+        return self._name
+
+    @cached_property
+    def human_readable(self):
+        """The memory estimate in human-readable format"""
+        return frozendict({k: humanbytes(v) for k, v in self.items()})
+
+    def __repr__(self):
+        return f'{self.__class__.__name__}({self.name}): {self.human_readable._dict}'
```

Comment:
You could have the MemoryEstimate take the op name as well so you can have a more explicit repr.